spyder/plugins/completion/plugin.py

```diff
@@ -50,6 +50,7 @@ class CompletionManager(SpyderCompletionPlugin):
             LSPRequestTypes.DOCUMENT_COMPLETION: {
                 KiteCompletionPlugin.COMPLETION_CLIENT_NAME,
                 LanguageServerPlugin.COMPLETION_CLIENT_NAME,
+                FallbackPlugin.COMPLETION_CLIENT_NAME
```

Comment:
@metalogical, I think you decided to not include fallback here. Do you remember why?

Reply:
After the `wait_for` timeout has elapsed, we return immediately once *any one* of the `WAIT_FOR_SOURCE`s has returned a completion.

If we add fallback here, then basically, we'll return immediately once `wait_for` timeout elapses regardless of whether or not either of LSP/Kite has returned completions.

This may be ok, but we decided to wait for one of those and ignore fallback when we first implemented this.

Reply:
Well, our tests are failing precisely for Kite, so this is not working correctly. Then how is it that fallback is working for Python files and not for non-Python files with no LSP nor Kite support? Adding this line here was the only way I found to make fallback work for those files.

Should we make LSP and Kite return an empty list (as suggested by @andfoy) to fix that?

Reply:
yes, the assumption is that Kite/LSP always return at least an empty list. I thought I fixed that when I implemented this, but I may have missed something. I also didn't consider the non-Python file case carefully.

Reply:
I found a way to avoid the waiting logic you added when fallback is the only client available.